hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSAdmin.java

```diff
@@ -632,6 +638,20 @@ private static void printDataNodeReports(DistributedFileSystem dfs,
     }
   }
 
+  private static void printSlowDataNodeReports(DistributedFileSystem dfs, boolean listNodes,
```

Comment:
Can you provide a sample output? It would be confusing, I guess. I suspect you would need some kind of header to distinguish from the other data node reports.

Reply:
> I suspect you would need some kind of header to distinguish from the other data node reports.

This is called only if condition `listAll || listSlowNodes` is true:
```
    if (listAll || listSlowNodes) {
      printSlowDataNodeReports(dfs, listSlowNodes, "Slow");
    }
```

Sample output:

<img width="524" alt="Screenshot 2022-03-25 at 9 12 58 PM" src="https://user-images.githubusercontent.com/34790606/165455352-303eb506-0a5f-491d-ac44-bcc243a8f0f6.png">

Reply:
One comment on the slow datanode report is that it seems to say nothing about why the NN thinks it slow; it does not note the attributes that are in excess of normal. Should it? (Hard to do when this info is not part of DatanodeInfo) For example, say something about how in excess a DNs latency is? (Perhaps this could be added later)

Reply:
> One comment on the slow datanode report is that it seems to say nothing about why the NN thinks it slow;

It's a datanode that determines whether it's peer datanodes are slower, NN just aggregates all DN reports.

> For example, say something about how in excess a DNs latency is? (Perhaps this could be added later)

Sure, this can be added as an additional info. Will create a follow-up Jira. Thanks @saintstack

Reply:
Fantastic. Thanks for offering the output screenshot.